elooplib/client.cpp

```diff
@@ -1,131 +1,187 @@
+  char buff[512];
+  size_t rcv;
+
+  rcv = recv(getFd(), buff, 512, 0);
+  if (rcv < 0)
+  {
+    return -1;
+  }
+  if (rcv == 0)
+  {
+    Node::closeConnection();
+    return 0;
+  }
+  int res = 0;
+  if (rcv > 0)
+  {
+    request_field.append(buff, rcv);
+    auto p = request_field.find('\n');
+    if (p != std::string::npos)
     {
-        request_field.append(buff, rcv); 
-        auto p = request_field.find('\n');
-        if (p != std::string::npos)
-        {
-
-            response = request_field.substr(0, p + 1);
-            sendData();
-            request_field.clear();
-        }
+
+      response = request_field.substr(0, p + 1);
+      res = sendData();
+      request_field.clear();
     }
-    return 0;
+  }
+  return res;
 }
 
-void Client::recognizeData()
+int Client::recognizeData()
 {
-    std::string filter{" \n"};
-
-    auto pos = response.find_first_of(filter);
-    std::string command = response.substr(0, pos + 1);
-
-    if (command == "time\n")
-    {
-        Client::handleTime();
-        return;
-    }
-
-    if (command == "echo ")
-    {
-        Client::handleEcho();
-        return;
-    }
+  if (response.starts_with("time\n"))
+  {
+    return Client::handleTime();
+  }
+
+  if (response.starts_with("echo "))
+  {
+    return Client::handleEcho();
+  }
+
+  if (response.starts_with("compress "))
+  {
+    return Client::handleCompress();
+  }
+  else
+  {
+    response = "Wrong request\n";
+    return 0;
+  }
+}
 
-    if (command == "compress ")
-    {
-        Client::handleCompress();
-        return;
-    }
-    else
-    {
-        response = "Wrong request\n";
-    }
+int Client::handleEcho()
+{
+  response = response.substr(5, request_field.size() - 5);
+  return 0;
 }
 
-void Client::handleEcho()
+int Client::handleTime()
 {
-    response = response.substr(5, request_field.size() - 5);
+  auto t = std::chrono::system_clock::now();
+  auto tt = std::chrono::system_clock::to_time_t(t);
+  response = std::ctime(&tt);
+  return 0;
 }
 
-void Client::handleTime()
+int Client::handleCompress()
+{
+  response = response.substr(9, request_field.size() - 10);
+  const std::filesystem::path dir(response);
+  if (!exists(dir))
+  {
+    response = "Path is not exists\n";
+    return 0;
+  }
+  int fd = eventfd(0, EFD_CLOEXEC);
+  return fd;
+}
+int Client::sendData()
 {
-    auto t = std::chrono::system_clock::now();
-    auto tt = std::chrono::system_clock::to_time_t(t);
-    response = std::ctime(&tt);
+  int r = recognizeData();
+  if (r == 0)
+  {
+    int s = send(getFd(), response.c_str(), response.size(), 0);
+    if (s < 0)
+      throw serverExcept("send()");
+    return r;
+  }
+  else // functions requires a threadpool
+  {
+    int s = send(getFd(), "Start compression...\n", 21, 0);
+    if (s < 0)
+      throw serverExcept("send()");
+    return r;
+  }
 }
 
-void Client::handleCompress()
+size_t Client::getFd() { Node::getFd(); };
+
+std::string Client::getResponse() { return response; };
```

Comment:
1. Extra `;`
2. Make this method `const`
3. I guess you want to return by reference:
`const std::string& Client::getResponse() const { return response; }`